File: entities-management/src/main/java/de/uni/stuttgart/ipvs/em/entities/dto/EntityCreateDto.java
package de.uni.stuttgart.ipvs.em.entities.dto;

import lombok.Data;

import de.uni.stuttgart.ipvs.em.entities.properties.HardwareProperties;
import de.uni.stuttgart.ipvs.em.entities.properties.LocationProperties;
import de.uni.stuttgart.ipvs.em.entities.properties.MessageProperties;
import de.uni.stuttgart.ipvs.em.entities.properties.TopicProperties;

@Data
public class EntityCreateDto
        implements TopicProperties, MessageProperties, HardwareProperties, LocationProperties {

    private String path;
    private String middlewareEndpoint;
    private String dataType;
    private String protocol;
    private String topicType;

    private String hardwareType;
    private String unit;

    private String messageFormat;
    private String metaModelType;
    private String metaModel;

    private String country;
    private String state;
    private String city;
    private String street;
    private String point;
}

File: cvi-web/src/app/entities/dto/entity-create-dto.ts
export interface EntityCreateDto {
  path: string;
  protocol: string;
  middlewareEndpoint: string;
  topicType: string;
  dataType: string;

  componentType: string;
  unit: string;

  messageFormat: string;
  metaModelType: string;
  metaModel: string;

  country: string;
  state: string;
  city: string;
  street: string;
}
